File: es_10.cpp
// 10. Inserire 10 numeri,
// compresi tra 10 e 20 (con controllo finchè non è compreso),
// calcolare il prodotto dei numeri inseriti.

#include <iostream>
using namespace std;

int main() {
    int n, prodotto = 1;

    for (int i = 1; i <= 10; i++) {
        do {
            cout << "Inserisci un numero compreso tra 10 e 20: ";
            cin >> n;
        } while (n < 10 || n > 20);

        prodotto *= n;
    }

    cout << "Il prodotto è: " << prodotto << endl;

    return 0;
}

File: Python/es_10.py
from itertools import count

# 10. Inserire 10 numeri,
# compresi tra 10 e 20 (con controllo finchè non è compreso),
# calcolare il prodotto dei numeri inseriti.

def chiedi_n() -> int:
    while True:
        n = int(
            input('Inserisci un numero tra 10 e 20: ')
        )
        
        if 10 <= n <= 20: break
        else: print('Numero non valido')
        
    return n

prod = 1
for i in count (start=1, step=1):
    n = chiedi_n()
    prod *= n
    
    if i >= 10: break

print(
    f'Il prodotto è: {prod}'
)